### PR TITLE
[Feature] Add paniptic segmentation evaluation

### PR DESCRIPTION
**Pull request by xizaoqu**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Support point cloud panoptic segmentation for semanticKITTI and nuScenes.

## Modification

Add panoptic segmentation evaluation and the test module.

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdet3d/evaluation/functional/panoptic_seg_eval.py [hunk 1]
```diff
@@ -0,0 +1,329 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+import numpy as np
+from mmengine.logging import print_log
+
+
+class EvalPanoptic:
```
Comment (ZwwWayne) [inline mmdet3d/evaluation/functional/panoptic_seg_eval.py:6]: We need to implement a PQ metric similar to the PQ metric in MMDet.

File: mmdet3d/evaluation/functional/panoptic_seg_eval.py [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+        offset (int): Offset for instance ids to concat with
+            semantic labels.
+        label2cat (tuple): Map from label to category.
+        ignore_index (list[int]): Ignored classes in evaluation.
+        logger (logging.Logger | str, optional): Logger used for printing.
+            Default: None.
+    """
+
+    def __init__(self,
+                 classes,
```
Comment (JingweiZhang12) [inline mmdet3d/evaluation/functional/panoptic_seg_eval.py:25]: Please add typehint for input arguments in these functions.

File: mmdet3d/evaluation/functional/panoptic_seg_eval.py (hunk 1 above)
Comment (JingweiZhang12) [inline mmdet3d/evaluation/functional/panoptic_seg_eval.py:6]: You can refer to [InstanceSegMetric](https://github.com/open-mmlab/mmdetection3d/blob/dev-1.x/mmdet3d/evaluation/metrics/instance_seg_metric.py) to implement the PanopticSegMetric in `mmdet3d/evaluation/metrics/panoptic_seg_metric.py` and some evaluation utils can be written in `mmdet3d/evaluation/functional/panoptic_seg_eval.py`

Comment (xizaoqu) [reply mmdet3d/evaluation/functional/panoptic_seg_eval.py:8]: OK, I've added type hints and a function wrapper for this util.

**Pull request closed without merging**